google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -397,7 +416,10 @@ protected Entity computeNext() {
           EXCEPTION_HANDLER,
           getOptions().getClock());
     } catch (RetryHelperException e) {
+      span.setStatus(Status.UNKNOWN.withDescription(e.getMessage()));
```

Comment:
Any specific reason for `Status.UNKNOWN` instead of `status.INTERNAL`? Generally, an example of  (UNKNOWN) where this error may be returned is if a Status value received from another address space belongs to an error-space that is not known in this address space. On the other side, Internal errors mean some invariants expected by underlying system has been broken. 

Also, `DEADLINE_EXCEEDED` can be useful in case of retries exceeded.

Having said that, I am not familiar with datastore codebase. Feel free to ignore if you disagree.